packages/terra-rpc-provider/lib/TerraRpcProvider.ts

```diff
@@ -135,4 +135,14 @@ export default class TerraRpcProvider extends NodeProvider implements FeeProvide
       }
     }
   }
+
+  async getTaxFees(amount: number, denom = 'uusd'): Promise<any> {
+    const taxRate = await this._lcdClient.treasury.taxRate()
+    const taxCap = await this._lcdClient.treasury.taxCap(denom)
+
+    const _taxRate = taxRate.toNumber()
+    const _taxCap = taxCap.amount.toNumber()
+
+    return Math.min((amount || 0) * _taxRate, _taxCap / 1_000_000)
```

Comment:
Shouldn't amount be always provided? Why the default to 0?

Reply:
Because initially Swap.vue pass undefined and evaluates to NaN 😄